javalanglib/src/main/scala/java/lang/Character.scala

```diff
@@ -282,8 +335,151 @@ object Character {
+        (64112 <= c && c <= 64217) || (131072 <= c && c <= 173782) ||
+        (173824 <= c && c <= 177972) || (177984 <= c && c <= 178205) ||
+        (194560 <= c && c <= 195101)
+  }
+
+  def isJavaIdentifierStart(ch: scala.Char): scala.Boolean =
+    isJavaIdentifierStart(ch.toInt)
+
+  def isJavaIdentifierStart(codePoint: scala.Int): scala.Boolean =
+    isJavaIdentifierStartImpl(getType(codePoint))
+
+  @inline
+  private[this] def isJavaIdentifierStartImpl(tpe: scala.Int): scala.Boolean = {
+    isLetterImpl(tpe) || tpe == LETTER_NUMBER || tpe == CURRENCY_SYMBOL ||
+      tpe == CONNECTOR_PUNCTUATION
+  }
+
+  def isJavaIdentifierPart(ch: scala.Char): scala.Boolean =
+    isJavaIdentifierPart(ch.toInt)
+
+  def isJavaIdentifierPart(codePoint: scala.Int): scala.Boolean =
+    isJavaIdentifierPartImpl(codePoint, getType(codePoint))
+
+  @inline private[this] def isJavaIdentifierPartImpl(codePoint: scala.Int,
+      tpe: scala.Int): scala.Boolean = {
+    isLetterImpl(codePoint) || tpe == CURRENCY_SYMBOL ||
+      tpe == CONNECTOR_PUNCTUATION || tpe == DECIMAL_DIGIT_NUMBER ||
+      tpe == LETTER_NUMBER || tpe == COMBINING_SPACING_MARK ||
+      tpe == NON_SPACING_MARK || isIdentifierIgnorableImpl(codePoint, tpe)
+  }
+
+  def isUnicodeIdentifierStart(ch: scala.Char): scala.Boolean =
+    isUnicodeIdentifierStart(ch.toInt)
+
+  def isUnicodeIdentifierStart(codePoint: scala.Int): scala.Boolean =
+    isUnicodeIdentifierStartImpl(getType(codePoint))
+
+  @inline
+  private[this] def isUnicodeIdentifierStartImpl(tpe: scala.Int): scala.Boolean =
+    isLetterImpl(tpe) || tpe == LETTER_NUMBER
+
+  def isUnicodeIdentifierPart(ch: scala.Char): scala.Boolean =
+    isUnicodeIdentifierPart(ch.toInt)
+
+  def isUnicodeIdentifierPart(codePoint: scala.Int): scala.Boolean =
+    isUnicodeIdentifierPartImpl(codePoint, getType(codePoint))
+
+  def isUnicodeIdentifierPartImpl(codePoint: scala.Int,
+      tpe: scala.Int): scala.Boolean = {
+    tpe == CONNECTOR_PUNCTUATION || tpe == DECIMAL_DIGIT_NUMBER ||
+      tpe == COMBINING_SPACING_MARK || tpe == NON_SPACING_MARK ||
+      isUnicodeIdentifierStartImpl(tpe) ||
+      isIdentifierIgnorableImpl(codePoint, tpe)
+  }
+
+  def isIdentifierIgnorable(c: scala.Char): scala.Boolean =
+    isIdentifierIgnorable(c.toInt)
+
+  def isIdentifierIgnorable(codePoint: scala.Int): scala.Boolean =
+    isIdentifierIgnorableImpl(codePoint, getType(codePoint))
+
+  @inline private[this] def isIdentifierIgnorableImpl(codePoint: scala.Int,
+      tpe: scala.Int): scala.Boolean = {
+    ('\u0000' <= codePoint && codePoint <= '\u0008') ||
+      ('\u000E' <= codePoint && codePoint <= '\u001B') ||
+      ('\u007F' <= codePoint && codePoint <= '\u009F') ||
+      tpe == FORMAT
+  }
+
+  def isMirrored(c: scala.Char): scala.Boolean =
+    isMirrored(c.toInt)
+
+  def isMirrored(codePoint: scala.Int): scala.Boolean = {
+    val idx = Arrays.binarySearch(isMirrorIndices, codePoint) + 1
+    (Math.abs(idx) & 1) != 0
```

Comment:
Hum, this should simply be `Arrays.binarySearch(isMirrorIndices, codePoint) >= 0`, shouldn't it?